src/polyswarmd/bounties.py

```diff
@@ -54,18 +53,21 @@ def calculate_commitment(account, verdicts):
     return int_from_bytes(nonce), int_from_bytes(commitment)
 
 
-@functools.lru_cache(maxsize=256)
-def substitute_ipfs_metadata(ipfs_uri, validate=AssertionMetadata.validate):
-    """Download metadata from IPFS and validate it against the schema.
+@cache.memoize(30)
+def substitute_ipfs_metadata(ipfs_uri, validate=AssertionMetadata.validate, ipfs_root=None, session=None):
```

Comment:
Will `session` always be the same object here? otherwise `memoize` won't work.

Reply:
Each greenlet has a different session, but those will remain the same.

Reply:
Though, now that we only call it at most once per event, it probably isn't as needed as it was before.

Reply:
Think it's fine then.